x/foundation/keeper/msg_server.go

```diff
@@ -10,6 +10,13 @@ import (
 
 const gasCostPerIteration = uint64(20)
 
+func canFoundationAuthorize(msgTypeURL string) bool {
+	urls := map[string]bool{
+		foundation.ReceiveFromTreasuryAuthorization{}.MsgTypeURL(): true,
+	}
+	return urls[msgTypeURL]
```

Comment:
```suggestion
	if foundation.ReceiveFromTreasuryAuthorization{}.MsgTypeURL() == msgTypeURL {
		return true
	}
	return false
```

Reply:
It checks whether x/foundation can handle the msg, and `urls` is the list of the msg which the module can handle.